src/API/Data/BuildingRelationshipsRepository.cs

```diff
@@ -0,0 +1,115 @@
+				TryFindBuildingRelationship(db, id));
+
+
+		internal static async Task<Result<BuildingRelationship, Error>> CreateBuildingRelationship(BuildingRelationshipRequest body)
+			=> await ExecuteDbPipeline("create a building relationship", db =>
+				ValidateRequest(db, body)
+				.Bind(_ => TryCreateBuildingRelationship(db, body))
+				.Bind(created => TryFindBuildingRelationship(db, created.Id))
+			);
+
+		internal static async Task<Result<BuildingRelationship, Error>> UpdateBuildingRelationship(HttpRequest req,BuildingRelationshipRequest body, int relationshipId)
+		{
+			return await ExecuteDbPipeline($"update building relationship {relationshipId}", db =>
+				ValidateRequest(db, body)
+				.Bind(_ => TryFindBuildingRelationship(db, relationshipId))
+                .Tap(existing => LogPrevious(req, existing))
+				.Bind(existing => TryUpdateBuildingRelationship(db, existing, body))
+				.Bind(_ => TryFindBuildingRelationship(db, relationshipId))
+			);
+		}
+
+		internal static async Task<Result<bool, Error>> DeleteBuildingRelationship(int relationshipId)
+		{
+			return await ExecuteDbPipeline($"delete building relationship {relationshipId}", db =>
+				TryFindBuildingRelationship(db, relationshipId)
+				.Bind(existing => TryDeleteBuildingRelationship(db, existing))
+			);
+		}
+
+		private static async Task<Result<BuildingRelationship, Error>> TryFindBuildingRelationship(PeopleContext db, int id)
+		{
+			var result = await db.BuildingRelationships
+				.Include(r => r.Unit)
+				.Include(r => r.Building)
+				.SingleOrDefaultAsync(r => r.Id == id);
+			return result == null
+				? Pipeline.NotFound("No building support relationship was found with the ID provided.")
+				: Pipeline.Success(result);
+		}
+
+		private static async Task<Result<BuildingRelationship, Error>> TryCreateBuildingRelationship(PeopleContext db, BuildingRelationshipRequest body)
+		{
+			var buildingRelationship = new BuildingRelationship
+			{
+				UnitId = body.UnitId,
+				BuildingId = body.BuildingId
+			};
+
+			db.BuildingRelationships.Add(buildingRelationship);
+			await db.SaveChangesAsync();
+			return Pipeline.Success(buildingRelationship);
+		}
+
+		private static async Task<Result<BuildingRelationshipRequest, Error>> ValidateRequest(PeopleContext db, BuildingRelationshipRequest body)
+		{
+			if (body.UnitId == 0 || body.BuildingId == 0)
+			{
+				return Pipeline.BadRequest("The request body was malformed, the unitId and/or buildingId field was missing.");
+			}
+			//Unit is already being checked/found in the Authorization step
+			if (await db.Buildings.AnyAsync(b => b.Id == body.BuildingId) == false)
+			{
+				return Pipeline.NotFound("The specified unit and/or building does not exist.");
+			}
+			if (await db.BuildingRelationships.AnyAsync(r => r.BuildingId == body.BuildingId && r.UnitId == body.UnitId))
+			{
+				return Pipeline.Conflict("The provided unit already has a support relationship with the provided building.");
+			}
```

Comment:
What happens here if I create a relationship, and then update it with the same properties?

Reply:
That's a good point. I'll fix this to check if the existing match is the same one that's being updated. If it is, we can just return an ok.

Reply:
Works for me. Thanks!